vsdenoise/mvtools/mvtools.py

```diff
@@ -45,990 +71,1092 @@ class MVTools:
-        )
-
-        if isinstance(search, SearchMode):
-            search = search(is_uhd=self.is_uhd, refine=self.refine, truemotion=motion.truemotion)
-
-        assert search
-
-        sad_mode = kwargs_fallback(  # type: ignore[assignment]
-            sad_mode, (self.analyze_func_kwargs, 'sad_mode'), SADMode.SATD
+        analyze_args = self.analyze_args | KwargsT(
+            blksize=blksize, blksizev=blksizev, levels=levels,
+            search=search, searchparam=searchparam, pelsearch=pelsearch,
+            lambda_=lambda_, chroma=self.chroma, truemotion=truemotion,
+            lsad=lsad, plevel=plevel, global_=global_,
+            pnew=pnew, pzero=pzero, pglobal=pglobal,
+            overlap=overlap, overlapv=overlapv, divide=divide,
+            badsad=badsad, badrange=badrange, meander=meander, trymany=trymany,
+            fields=self.fieldbased.is_inter, tff=self.fieldbased.is_tff, dct=dct
         )
 
-        vectors = MotionVectors() if inplace else self.vectors
+        if self.mvtools is MVToolsPlugin.INTEGER and not any(
+            (analyze_args.get('overlap'), analyze_args.get('overlapv'))
+        ):
+            self.disable_compensate = True
 
-        if isinstance(sad_mode, tuple):
-            sad_mode, recalc_sad_mode = sad_mode
+        if self.mvtools is MVToolsPlugin.FLOAT:
+            self.vectors.vmulti = self.mvtools.Analyze(super_clip, radius=self.tr, **analyze_args)
         else:
-            sad_mode, recalc_sad_mode = sad_mode, SADMode.SATD
-
-        supers = supers or self.get_supers(ref, inplace=inplace)
-
-        thSAD_recalc = thSAD
-
-        t2 = (self.tr * 2 if self.tr > 1 else self.tr) if self.source_type.is_inter else self.tr
-
-        analyze_args = KwargsT(
-            dct=sad_mode, pelsearch=search.pel, blksize=blocksize, overlap=overlap, search=search.mode,
-            truemotion=motion.truemotion, searchparam=search.param, chroma=self.chroma,
-            plevel=motion.plevel, pglobal=motion.pglobal, pnew=motion.pnew,
-            lambda_=motion.block_coherence(blocksize), lsad=motion.sad_limit,
-            fields=self.source_type.is_inter
-        ) | self.analyze_args
-
-        if self.mvtools is MVToolsPlugin.FLOAT_NEW:
-            vectors.vmulti = self.mvtools.Analyse(supers.search, radius=t2, **analyze_args)
-        else:
-            for i in range(1, t2 + 1):
-                vectors.calculate_vectors(i, self.mvtools, supers, False, self.finest_mode, **analyze_args)
-
-        if self.refine:
-            self.recalculate(
-                self.refine, self.tr, blocksize, halfoverlap, thSAD_recalc,
-                search, recalc_sad_mode, motion, vectors, supers, ref=ref
-            )
-
-        vectors.kwargs.update(thSAD=thSAD)
-
-        return vectors
+            for i in range(1, self.tr + 1):
+                for direction in MVDirection:
+                    vector = self.mvtools.Analyze(super_clip, isb=direction - 1, delta=i, **analyze_args)
```

Comment:
any reason not to use a more explicit check? `direction-1` does produce 0/1 as you want here, but still
```suggestion
                    vector = self.mvtools.Analyze(super_clip, isb=direction is MVDirection.BACK, delta=i, **analyze_args)
```